File: Exemplos SmartPOS/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_SmartPOS_Flutter-font/lib/IntentDigitalHubService/ELGINPAY/Commands/inicia_cancelamento_venda.dart
import 'package:flutter_smartpos/IntentDigitalHubService/ELGINPAY/elginpay_command.dart';

class IniciaCancelamentoVenda extends ElginPayCommand {
  final String _valorTotal;
  final String _ref;
  final String _data;

  IniciaCancelamentoVenda(this._valorTotal, this._ref, this._data)
      : super('iniciaCancelamentoVenda');

  @override
  get functionParametersJson => {
        'valorTotal': _valorTotal,
        'ref': _ref,
        'data': _data,
      };
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_SmartPOS_Ionic-font/src/intentDigitalHubService/ELGINPAY/Commands/IniciaCancelamentoVenda.ts
import {ElginPayCommand} from '../ElginPayCommand';

export class IniciaCancelamentoVenda extends ElginPayCommand {
  readonly valorTotal: string;
  readonly data: string;
  readonly ref: string;

  constructor(valorTotal: string, data: string, ref: string) {
    super('iniciaCancelamentoVenda');
    this.valorTotal = valorTotal;
    this.data = data;
    this.ref = ref;
  }

  functionParameters(): object {
    return {
      valorTotal: this.valorTotal,
      data: this.data,
      ref: this.ref,
    };
  }
}
